src/charm.py

```diff
@@ -191,6 +194,29 @@ def _on_update_status(self, _: ops.UpdateStatusEvent) -> None:
 
         self.unit.status = self._remove_unlisted_plugins(container=container)
 
+    def _on_jenkins_home_storage_attached(self, event: ops.StorageAttachedEvent) -> None:
+        """Correctly set permission when storage is attached.
+
+        Args:
+            event: The event fired when the storage is attached.
+        """
+        container = self.unit.get_container(self.state.jenkins_service_name)
+        if not container.can_connect():
+            event.defer()
```

Comment:
Just as a note, the current team guidelines are to avoid using `event.defer` if possible. cc @gregory-schiano

Reply:
Since `event.defer` is used consistently in this charm, moving the logic from `event.defer` to a reconciliation loop imho requires a refactoring discussion and probably a follow up PR. I'm marking this conversation as resolved but let's keep tracking this conversation